Remotion/Validation/Core/Attributes/Validation/LengthValidationAttribute.cs

```diff
@@ -29,50 +29,69 @@ namespace Remotion.Validation.Attributes.Validation
   /// </summary>
   public class LengthValidationAttribute : AddingValidationAttributeBase
   {
-    // TODO RM-5906: make max-length nullable and create specific type of length-validator based on min and max-length values
-    private readonly int _maxLength;
-    private readonly int _minLength;
-
     /// <summary>
-    /// Instantiates a new <see cref="LengthValidationAttribute"/>.
+    /// Gets or sets the minimum number of characters required.
     /// </summary>
-    /// <param name="minLength">The minimum number of characters required.</param>
-    /// <param name="maxLength">The maximum number of characters allowed.</param>
-    public LengthValidationAttribute (int minLength, int maxLength)
-    {
-      _minLength = minLength;
-      _maxLength = maxLength;
-    }
+    public int MinLength { get; set; }
 
-    public int MinLength
-    {
-      get { return _minLength; }
-    }
+    /// <summary>
+    /// Gets or sets the maximum number of characters allowed.
+    /// </summary>
+    public int MaxLength { get; set; }
 
-    public int MaxLength
+    /// <summary>
+    /// Instantiates a new <see cref="LengthValidationAttribute"/>.
+    /// </summary>
+    public LengthValidationAttribute ()
     {
-      get { return _maxLength; }
     }
 
     protected override IEnumerable<IPropertyValidator> GetValidators (IPropertyInformation property, IValidationMessageFactory validationMessageFactory)
     {
       ArgumentUtility.CheckNotNull("property", property);
       ArgumentUtility.CheckNotNull("validationMessageFactory", validationMessageFactory);
 
-      LengthValidator validator;
+      IPropertyValidator validator;
       if (string.IsNullOrEmpty(ErrorMessage))
       {
         validator = PropertyValidatorFactory.Create(
             property,
-            parameters => new LengthValidator(MinLength, MaxLength, parameters.ValidationMessage),
+            parameters => CreateValidator(parameters.ValidationMessage),
             validationMessageFactory);
       }
       else
       {
-        validator = new LengthValidator(MinLength, MaxLength, new InvariantValidationMessage(ErrorMessage));
+        validator = CreateValidator(new InvariantValidationMessage(ErrorMessage));
       }
 
       return EnumerableUtility.Singleton(validator);
     }
+
+    private IPropertyValidator CreateValidator (ValidationMessage message)
+    {
+      if (MinLength != 0 && MaxLength > MinLength)
```

Comment:
```suggestion
          private IPropertyValidator CreateValidator (ValidationMessage message)
    {
      if (MinLength == 0 && MaxLength > 0)
        return new MaximumLengthValidator(MaxLength, message);
      else if (MaxLength == 0 & MinLength > 0)
        return new MinimumLengthValidator(MinLength, message);
      else if (MinLength > 0 && MaxLength > 0 && MinLength == MaxLength)
        return new ExactLengthValidator(MinLength, message);
      else
        return new LengthValidator(MinLength, MaxLength, message);
    }
```
I would prefer here to use the if/else instead of the switch expression because this way it is more readable. Now, it is possible to get rid of the exception since the lengthvalidator is the default. If an incorrect value (<1) gets assigned, it should throw the exception in LengthValidator

What do you think?

Reply:
That's what I was looking for / expecting :)

Reply:
Ah yeah, thanks :)

Reply:
Just because I wanted to try how this might look:

![image](https://github.com/re-motion/Framework/assets/58393218/034bdb00-0b39-4e8e-acff-6b53eeca79cd)

It might be a bit unfitting in re-motion though, so I also agree with taking the if-else blocks 👍

Reply:
Thanks @Frazerus, for the demo!

My thoughts: since this check relies only on properties of the switched-on value, not the actual value itself (which would be `this` in our case), it feels idiomatically wrong to use a switch expression. The main reason for using switch is, that it's more compact but the resulting syntax, in particular the fact that we start with empty braces, shows to me that we're misusing a language feature (it looks awkward).

That said, there is also pattern matching which is implemented by switch:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#positional-pattern
https://www.thomasclaudiushuber.com/2021/02/25/c-9-0-pattern-matching-in-switch-expressions/
```
return (MinLength, MaxLength) switch
{
    (0, > 0) => new MaxiumLengthValidator(MaxLength),
    (> 0, 0) => new MinimumLengthValidator(MinLength),
    (_, MaxLength) => new ExactLengthValidator(MaxLength), // not 100% sure on the syntax
    _ => new LengthValidator(MinLength, MaxLength)
}
```
It's a very fine difference in the use of syntax elements compared to the simple application of using empty braces and `when`. However, idiomatic code often is :)

Another possible way to write this could be
```
return this switch
{
    { MinLength: 0, MaxLength: > 0 } => new MaxiumLengthValidator(MaxLength),
    { MinLength: > 0, MaxLength: 0 } => new MinimumLengthValidator(MinLength),
    { MinLength: MaxLength } => new ExactLengthValidator(MaxLength), // not 100% sure on the syntax
    _ => new LengthValidator(MinLength, MaxLength)
}
```
Not sure which is best.

Reply:
Yeah, I see how this might be problematic, I think we should however talk about switch expressions more in-depth at some point (dev-jour-fixe?). 

I think they can be really useful and I haven't seen much code written with them, making it harder to know what you can and cannot do.